homeassistant/components/sensibo/climate.py

```diff
@@ -141,44 +144,63 @@ async def async_assume_state(service):
 class SensiboClimate(ClimateEntity):
     """Representation of a Sensibo device."""
 
-    def __init__(self, client, data, units):
-        """Build SensiboClimate.
-
-        client: aiohttp session.
-        data: initially-fetched data.
-        """
+    def __init__(self, client: SensiboClient, data: dict[str, Any], units: str) -> None:
+        """Initiate SensiboClimate."""
         self._client = client
         self._id = data["id"]
         self._external_state = None
         self._units = units
-        self._available = False
-        self._do_update(data)
         self._failed_update = False
+        self._attr_available = False
+        self._attr_unique_id = self._id
+        self._attr_temperature_unit = (
+            TEMP_CELSIUS if data["temperatureUnit"] == "C" else TEMP_FAHRENHEIT
+        )
+        self._do_update(data)
+        self._attr_target_temperature_step = (
+            1 if self.temperature_unit == units else None
+        )
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, self._id)},
-            name=self._name,
+            name=self._attr_name,
             manufacturer="Sensibo",
             configuration_url="https://home.sensibo.com/",
             model=data["productModel"],
             sw_version=data["firmwareVersion"],
             hw_version=data["firmwareType"],
-            suggested_area=self._name,
+            suggested_area=self._attr_name,
         )
 
-    @property
-    def supported_features(self):
-        """Return the list of supported features."""
-        return self._supported_features
-
-    def _do_update(self, data):
-        self._name = data["room"]["name"]
-        self._measurements = data["measurements"]
+    def _do_update(self, data) -> None:
+        self._attr_name = data["room"]["name"]
         self._ac_states = data["acState"]
-        self._available = data["connectionStatus"]["isAlive"]
+        self._attr_extra_state_attributes = {
+            "battery": data["measurements"].get("batteryVoltage")
```

Comment:
Battery should be a separate sensor.

Reply:
@gjohansson-ST While this is existing, we have `SensorDeviceClass.BATTERY` on the `sensor` platform that is a better fit for this.

There is an example in `august/sensor.py`, `sonos/sensor.py`, etc

Reply:
It was there from before. Would not move it to a sensor make it a breaking change?

Reply:
Home Assistant is moving to creating separate sensors for almost all extra state attributes since they can be better represented this way and have attributes since as a device class which makes it easier for other integrations to consume.

A breaking change is desirable here for this reason.